packages/decorators/src/piece-decorators.ts

```diff
@@ -49,14 +49,21 @@ import { createClassDecorator, createProxy } from './utils';
  * }
  * ```
  */
-export function ApplyOptions<T extends PieceOptions>(optionsOrFn: T | ((container: Container, context: PieceContext) => T)): ClassDecorator {
+export function ApplyOptions<T extends PieceOptions>(
+	optionsOrFn: T | (({ container, context }: ApplyOptionsCallbackParameters) => T)
```

Comment:
Why are you destructuring these? How is this even valid o.o

Reply:
Hi I'm silly. How are you?